### PR TITLE
remove regex dependency

### PR DESCRIPTION
**Pull request by planetis-m**
This commit removes the dependency to regex library.

### CODE REVIEW

File: src/haloniumpkg/utils.nim [hunk 1]
```diff
@@ -113,20 +113,19 @@ proc `%`*(obj: tuple): JsonNode =
...
-      raise newException(URLTemplateException, fmt"Key '{key}' not found in JsonData")
-    if node[key].kind != JString:
-      raise newException(URLTemplateException, fmt"Key '{key}' is not a string")
-
-    vals.add(("$" & key, node[key].getStr()))
-
-  result = str.multiReplace(vals)
+  var keyVals: seq[(string, string)]
+  var isKey = false
+  for x in tokenize(str, {'$', '/'}):
```
Comment (jyapayne) [inline src/haloniumpkg/utils.nim:118]: What is the slash here for?

**Review by jyapayne — CHANGES_REQUESTED** [review]
Cool, thanks for this! Please also remove the dependency in the nimble file. I also have one question comment in the review.

File: src/haloniumpkg/utils.nim (hunk 1 above)
Comment (planetis-m) [reply src/haloniumpkg/utils.nim:118]: Perhaps I can explain it better with code:

```nim
import strutils

var keys: seq[string]
var found = false
for x in tokenize("/session/$sessionId/moz/xbl/$elementId/anonymous_by_attribute", {'$', '/'}):
  if x.isSep and x.token.endsWith('$'):
    found = true
  elif found and not x.isSep:
    found = false
    keys.add x.token

echo keys
```

It splits the path using both '$' and '/'

Comment (jyapayne) [reply src/haloniumpkg/utils.nim:118]: Ah, yep, got it!

**Pull request merged**